File: Core kernel razonamiento repo para Yaiwes/GitHub-Copilot-SDK/GitHub-Copilot-SDK/java/sdk/src/main/java/com/github/copilot/rpc/BearerTokenProvider.java
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/

package com.github.copilot.rpc;

import java.util.concurrent.CompletableFuture;

import com.github.copilot.CopilotExperimental;

/**
 * Functional interface for supplying per-provider bearer tokens for BYOK
 * provider requests.
 * <p>
 * The callback returns the raw token without a {@code Bearer } prefix. The SDK
 * keeps this callback client-side and the runtime requests a token via the
 * session-scoped {@code providerToken.getToken} RPC before each outbound model
 * request.
 * <p>
 * <strong>Experimental.</strong> This managed-identity surface may change or be
 * removed in future SDK or CLI releases.
 *
 * @see ProviderConfig#setBearerTokenProvider(BearerTokenProvider)
 * @see NamedProviderConfig#setBearerTokenProvider(BearerTokenProvider)
 * @since 1.0.0
 */
@CopilotExperimental
@FunctionalInterface
public interface BearerTokenProvider {

    /**
     * Gets a bearer token for the provider identified by {@code args}.
     *
     * @param args
     *            the provider token request arguments
     * @return a future that completes with the raw token, without a {@code Bearer }
     *         prefix
     */
    CompletableFuture<String> getToken(ProviderTokenArgs args);
}

File: ➡️📂Core kernel Yaiwes/GitHub-Copilot-SDK/GitHub-Copilot-SDK/dotnet/src/BearerTokenProvider.cs
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/

using System.Diagnostics.CodeAnalysis;

namespace GitHub.Copilot;

/// <summary>
/// Arguments passed to a bearer-token callback (the <c>BearerTokenProvider</c> property
/// on <see cref="ProviderConfig"/> / <see cref="NamedProviderConfig"/>) when the
/// runtime needs a fresh bearer token for a BYOK provider.
/// </summary>
/// <remarks>
/// Part of the experimental managed-identity / bearer-token-provider surface and
/// may change or be removed in future SDK or CLI releases.
/// </remarks>
[Experimental(Diagnostics.Experimental)]
public sealed class ProviderTokenArgs
{
    /// <summary>
    /// Name of the BYOK provider needing a token. For the singular, whole-session
    /// <see cref="ProviderConfig"/> this is the implicit provider name
    /// (<c>"default"</c>); for <see cref="NamedProviderConfig"/> entries it is
    /// <see cref="NamedProviderConfig.Name"/>.
    /// </summary>
    /// <remarks>
    /// The callback closes over its own token scope/audience; the runtime is
    /// provider-agnostic and forwards only the provider name.
    /// </remarks>
    public required string ProviderName { get; init; }

    /// <summary>
    /// Id of the session that triggered this token request. A client-level
    /// shared callback registered for many sessions can use this to resolve the
    /// owning session and scope token acquisition or caching per session.
    /// </summary>
    public required string SessionId { get; init; }
}
